8349702: jdk.internal.net.http.Http2Connection::putStream needs to provide cause while cancelling stream

vy opened the pull request:
Fixes stream cancellation in `Http2Connection::putStream`. See comments in the ticket for details on the issue. `tier1,2` results are attached to the ticket too.

jaikiran commented:
Hello Volkan, so from what I understand, before this change, we used to raise a generic `IOException` if a new request couldn't be initiated because the `Http2Connection` on which it is being raised already had some failure. With this new change, we propagate any original exception that was registered on the `Http2Connection` thus allowing any special handling of that original exception type before it gets propagated to the application code. Did I understand this correctly? 

The linked JBS issue has a `noreg-self`. I think it should instead be `noreg-hard`, since `noreg-self` is used when we change a regression test itself, which isn't the case here.

vy commented:
@jaikiran, thanks so much for the review, much appreciated! :bow: As discussed with you and @dfuch,

- Changed the ticket labels to `httpclient` and `intermittent`
- Added the JBS issue ID to the `@bug` list of `AsyncExecutorShutdown`
- Wrapped the `cause` in an `IOException` in `putStream()`

jaikiran reviewed (approved):
Thank you for these updates. This looks good to me. Since it was the `test/jdk/java/net/httpclient/AsyncExecutorShutdown.java` test which caught this intermittent failure, it's the right thing to update the `@bug` of this test to refer to this issue.

dfuch reviewed (approved):
Thanks Volkan! LGTM.

vy commented:
/integrate

jaikiran commented:
/sponsor
